Command/winzouStateMachineDebugCommand.php

```diff
@@ -123,4 +124,32 @@ protected function printTransitions(array $transitions, OutputInterface $output)
 
         $table->render();
     }
+
+    /**
+     * @param array           $allCallbacks
+     * @param OutputInterface $output
+     */
+    protected function printCallbacks(array $allCallbacks, OutputInterface $output)
+    {
+        foreach ($allCallbacks as $type => $callbacks) {
+            if ($type === 'before' || $type === 'after') {
+                $table = new Table($output);
+                $table->setHeaders(array(ucfirst($type) . ' Callback', 'On', 'Do', 'Args'));
+
+                end($callbacks);
+                $lastCallback = key($callbacks);
+                reset($callbacks);
+
+                foreach ($callbacks as $name => $callback) {
+                    $table->addRow(array($name, implode("\n", $callback['on']), implode("\n", $callback['do']), implode("\n", $callback['args'])));
+
+                    if ($name !== $lastCallback) {
+                        $table->addRow(new TableSeparator());
```

Comment:
Is it necessary? I'm afraid the table can be quite long if you have a lot of callbacks and use a separator between each. What do you think?

Reply:
I think without separators between callbacks, huge tables won't be easily readable.

Take a look at this gist : https://gist.github.com/PapsOu/34778be7cf29c2e931283009942e6cb6

If you prefer, I can add an option to display callbacks tables only if specified ?

Reply:
Here the gist without separators : https://gist.github.com/PapsOu/d5f1226bd2a6a03a69e87968f2199ecc